examples/qml_minimal/CMakeLists.txt

```diff
@@ -22,125 +22,208 @@ set(CMAKE_AUTOMOC ON)
+    # overwriting another's files.
+    target_include_directories(${APP_NAME}_lib INTERFACE "${CXXQT_EXPORT_DIR}/${CRATE}")
+
+    target_link_libraries(${APP_NAME}_lib INTERFACE
+        # WHOLE_ARCHIVE is needed for the generated QML plugin to register on startup,
+        # otherwise the linker will discard the static variables that initialize it.
+        "$<LINK_LIBRARY:WHOLE_ARCHIVE,${CRATE}-static>"
+        Qt::Core
+        Qt::Gui
+        Qt::Qml
+        Qt::QuickControls2
     )
+    # ANCHOR_END: book_cmake_use_corrosion
 
-    # RUNTIME_ENV comes from the CMakeLists.txt at the root of this repository.
-    set_tests_properties(
-        ${APP_NAME}_${TEST_NAME}_test PROPERTIES ENVIRONMENT_MODIFICATION
-                                                 "${RUNTIME_ENV}"
-    )
-    if (COMMAND add_valgrind_test)
-       add_valgrind_test(
-          ${APP_TEST_NAME} "${TEST_CMD}" ${CMAKE_CURRENT_BINARY_DIR}
-       )
+    # ANCHOR: book_cmake_executable
+    # Define the executable with the C++ source
+    add_executable(${APP_NAME} cpp/main.cpp)
+
+    # Link to the Rust library
+    target_link_libraries(${APP_NAME} PRIVATE ${APP_NAME}_lib)
+
+    # If we are using a statically linked Qt then we need to import any qml plugins
+    qt_import_qml_plugins(${APP_NAME})
+    # ANCHOR_END: book_cmake_executable
+
+    #
+    # Unit test
+    #
+
+    if(TARGET Qt6::Core)
+        find_package(Qt6 COMPONENTS QuickTest Test REQUIRED)
     else()
-       MESSAGE(STATUS "add_valgrind_test is defined in the top level of CXX-Qt. It will not executed")
+        find_package(Qt5 COMPONENTS QuickTest Test REQUIRED)
     endif()
-endfunction()
 
-add_qml_test(myobject)
+    function(add_qml_test TEST_NAME)
+        set(APP_TEST_NAME ${APP_NAME}_${TEST_NAME}_test)
+        add_executable(${APP_TEST_NAME} tests/${TEST_NAME}/tst_${TEST_NAME}.cpp)
+        target_link_libraries(${APP_TEST_NAME} PRIVATE
+            ${APP_NAME}_lib
+            Qt::QuickTest
+        )
+        qt_import_qml_plugins(${APP_TEST_NAME})
+
+        set(TEST_CMD
+            $<TARGET_FILE:${APP_TEST_NAME}> -input
+            ${CMAKE_CURRENT_SOURCE_DIR}/tests/${TEST_NAME}/tst_${TEST_NAME}.qml
+        )
+
+        add_test(
+            NAME ${APP_TEST_NAME}
+            COMMAND ${TEST_CMD}
+        )
+
+        # RUNTIME_ENV comes from the CMakeLists.txt at the root of this repository.
+        set_tests_properties(
+            ${APP_NAME}_${TEST_NAME}_test PROPERTIES ENVIRONMENT_MODIFICATION
+                                                    "${RUNTIME_ENV}"
+        )
+        if (COMMAND add_valgrind_test)
+            add_valgrind_test(
+                ${APP_TEST_NAME} "${TEST_CMD}" ${CMAKE_CURRENT_BINARY_DIR}
+            )
+        else()
+            MESSAGE(STATUS "add_valgrind_test is defined in the top level of CXX-Qt. It will not executed")
+        endif()
+    endfunction()
+
+    add_qml_test(myobject)
```

Comment:
then all of the test stuff could go behind a new `BUILD_TESTS` option, although wonder what happens if you try to build the example standalone and it has a `if(BUILD_TESTS)` but that option is defined in the root CMake :thinking: